packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -530,12 +530,18 @@ export class EditorPage {
 		if ( envVariables.VIEWPORT_NAME === 'desktop' ) {
 			await this.editorBlockToolbarComponent.clickParentBlockButton( expectedParentBlockName );
 		} else {
-			await this.editorBlockToolbarComponent.clickOptionsButton();
+			// If the menu was already open due to another action, don't open it again.
+			if ( ! ( await this.editorBlockToolbarComponent.isOptionsMenuOpen() ) ) {
+				await this.editorBlockToolbarComponent.clickOptionsButton();
+			}
 			await this.editorPopoverMenuComponent.clickMenuButton(
 				`Select parent block (${ expectedParentBlockName })`
 			);
-			// It stays open on modal! We have to close it again.
-			await this.editorBlockToolbarComponent.clickOptionsButton();
```

Comment:
I could see when running the test that the comment wasn't true, the menu does close after selecting the parent block, and this line was reopening it.

The code now checks to see whether the menu is open to be sure.